tests/lean/run/json_empty.lean

```diff
@@ -0,0 +1,52 @@
+import Lean
+
+/-!
+# Validation of `Json Empty` instance
+
+This test serves a similar purpose to the `Repr Empty` test in `repr_empty.lean`.
+-/
+
+structure Foo (α : Type) where
+  y : Option α
+deriving Lean.ToJson, Lean.FromJson, Repr
+
+def fromStrHelper (res : Type) [Lean.FromJson res] (s : String) : Except String res :=
+  (Lean.Json.parse s) >>= Lean.fromJson?
+
+/-! First, two basic examples with α not-Empty -/
+/-- info: {"y": 1} -/
+#guard_msgs in
+#eval Lean.toJson (⟨some 1⟩ : Foo Nat)
+
+/-! Ensure we can parse the type back -/
+/-- info: Except.ok { y := some 1 } -/
+#guard_msgs in
+#eval fromStrHelper (Foo Nat) "{\"y\": 1}"
```

Comment:
Perhaps it's not in core, but somewhere there's a `json%` elaborator which should avoid the quoting awkwardness here.

Reply:
There's also raw strings:
```suggestion
#eval fromStrHelper (Foo Nat) r#"{"y": 1}"#
```

Reply:
```suggestion
#eval Lean.fromJson? (α := Foo Nat) <| json% {"y": 1}
```

Reply:
@eric-wieser 
Thanks for the suggestion!  I will change.  Initially, I was trying to keep the test "as simple as possible", i.e. without involving additional Elab machinery such as `json%`

@david-christiansen 
Thanks, TIL.  I've not come across raw strings in FPiL yet 😄  FWIW, this (esp the PR preceding this one) is inspired because I wanted to `#eval`/`Repr` the `inductive Prim (special : Type)` in Section 5.2, so thanks for the motivation!

Reply:
(Raw strings are less than a year old — you could take a look at #2929, which also included a bit of documentation about the lexical structure.)

Reply:
Done.

Reply:
Glad you liked the book! I hope to find time soon to update it for all the new things that Lean has become able to do since way back in 4.3 :-)